tensorflow/stream_executor/rocm/rocm_blas.h

```diff
@@ -97,16 +98,48 @@ class ROCMBlas : public blas::BlasSupport {
                               /*err_on_failure=*/false, args...);
   }
 
+  // A helper struct to for type conversion in DoBlasGemmBatchedInternal
+  template <typename T>
+  struct EigenHalfToRocBlasHalf{
+    using type = T;
+  };
+  
+  // A helper allocation funciton to convert raw pointers memory layout to
+  // strided flavor
+  template <typename T>
+  port::Status AllocateStridedBuffer(
+      const std::vector<typename EigenHalfToRocBlasHalf<T>::type *> &raw_ptrs,
```

Comment:
@jerryyin reviewing the interface makes me wonder the meaning of `EigenHalfToRocBlasHalf`, and should we rename it.

1) a new name for `EigenHalfToRocBlasHalf` would be desired, as `AllocateStridedBuffer` would take any type T as its template argument, not necessarily restricted to Half types, but the interface declared here may make one wonder it is only applicable to Half types.
2) specialization of `EigenHalfToRocBlasHalf` is in `.cpp` not `.h` makes it harder to grasp the meaning of `EigenHalfToRocBlasHalf`.
3) `MAPPED_T` is only used inside the implementation of `AllocateStridedBuffer` while it could be used to help simplify the declaration here.
4) the true meaning of `EigenHalfToRocBlasHalf` is actually mentioned in the comment of `MAPPED_T`. it shall be put in the declaration of `EigenHalfToRocBlasHalf` above.

Reply:
Talked offline:
1. `EigenHalfToRocBlasHalf` -> `RocBlasTypeConversionHelper`
2. Moved the specialization to cpp
3. Compiler cannot correctly resolve the typedef of template, possibly due to reason [here](https://stackoverflow.com/questions/5845950/can-i-typedef-a-template-template-parameter).
4. Added comments above `RocBlasTypeConversionHelper`. Removed MAPPED_T comments from cpp file because multiple places are using this typedef.